DB-4830: Set Pantheon Environment Env Variables (#434)
pantheon-systems/decoupled-kit-js: web/docs/Frontend Starters/Next.js/Next.js + WordPress/setting-environment-variables.md

can be used inside of `next.config.js`.

- `IS_LIVE_ENVIRONMENT` - Automatically set to true if
  `PANTHEON_ENVIRONMENT_URL` is live and false otherwise

Either the `PANTHEON_CMS_ENDPOINT` or `WPGRAPHQL_URL` will need to be set.

The `PANTHEON_CMS_ENDPOINT` can be mocked for local development by defining it
in the `.env.development.local`

```
PANTHEON_CMS_ENDPOINT=
```

Taking a look at how the `next.config.js` works, there is this logic which sets
the `backendUrl`

```
let backendUrl, imageDomain;
if (process.env.WPGRAPHQL_URL === undefined) {
	backendUrl = `https://${process.env.PANTHEON_CMS_ENDPOINT}/wp/graphql`;
	imageDomain = process.env.IMAGE_DOMAIN || process.env.PANTHEON_CMS_ENDPOINT;

	// populate WPGRAPHQL_URL as a fallback and for build scripts
	process.env.WPGRAPHQL_URL = `https://${process.env.PANTHEON_CMS_ENDPOINT}/wp/graphql`;
} else {
	backendUrl = process.env.WPGRAPHQL_URL;
	imageDomain =
		process.env.IMAGE_DOMAIN ||
		process.env.WPGRAPHQL_URL.replace(/\/wp\/graphql$/, '').replace(
			/^https?:\/\//,
			'',
		);
}
```

In order to connect to a Multidev backend, this backendUrl will need to be

> updated. `PANTHEON_ENVIRONMENT_URL` has the data needed to source your backend
> and will need to be parsed according to your needs.

Reviewer: ```suggestion
updated. `PANTHEON_ENVIRONMENT_URL` includes the PR number or integration branch name
and will need to be parsed according to your needs.
```
Doesn't have to be the suggestion, but I think we need something less vague than 'has the data needed'

IMAGE_DOMAIN=my-image-cdn.site
```

> ## Connecting to Multidev Cloud Environments

Reviewer: ```suggestion
## Connecting to Multidev Environments
```

To connect to a Multidev environment, the following helper environment variable
can be used inside of `next.config.js`.

> - `IS_LIVE_ENVIRONMENT` - Automatically set to true if
>   `PANTHEON_ENVIRONMENT_URL` is live and false otherwise

Reviewer: ```suggestion
- `IS_LIVE_ENVIRONMENT` - True if
  `PANTHEON_ENVIRONMENT_URL` is live.
```

Either the `PANTHEON_CMS_ENDPOINT` or `WPGRAPHQL_URL` will need to be set.

The `PANTHEON_CMS_ENDPOINT` can be mocked for local development by defining it
in the `.env.development.local`

```

> PANTHEON_CMS_ENDPOINT=

Reviewer: Add an example here. This is needed because the `PANTHEON_CMS_ENDPOINT` is not the same format as the `WPGRAPHQL_URL`
```suggestion
PANTHEON_CMS_ENDPOINT=dev-my-wordpress-site.pantheonsite.io
```
